### PR TITLE
[SYCLomatic] Handle typedefs to `CUDA_RESOURCE_DESC`

### PR DESCRIPTION
**Pull request by the-slow-one**
Migration of `CUDA_RESOURCE_DESC` breaks if the struct is "typedef"ed by the user. See below.

The following code migrates as expected
```
#include <cuda.h>

CUDA_RESOURCE_DESC works(CUarray arr) {
  CUDA_RESOURCE_DESC rdesc;
  rdesc.resType = CU_RESOURCE_TYPE_ARRAY;
  rdesc.res.array.hArray = arr;
  return rdesc;
}
```
Whereas the code below doesn't migrate as intended.
```
typedef CUDA_RESOURCE_DESC ATYPE;
ATYPE donot_work(CUarray arr, cudaResourceDesc a) {
  ATYPE rdesc;
  rdesc.resType = CU_RESOURCE_TYPE_ARRAY;
  rdesc.res.array.hArray = arr;
  return rdesc;
}                                                                                                                                                                                                                                                                                
```
This PR fixes this issue.

Signed off by: [Deepak Raj H R](deepak.raj.h.r@intel.com)

### CODE REVIEW

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -13099,10 +13099,10 @@ void TextureRule::registerMatcher(MatchFinder &MF) {
                             "cudaTextureObject_t", "CUtexObject"))))))
                     .bind("texObj"),
                 this);
-  MF.addMatcher(memberExpr(hasObjectExpression(hasType(namedDecl(hasAnyName(
+  MF.addMatcher(memberExpr(hasObjectExpression(hasType(type(hasUnqualifiedDesugaredType(recordType(hasDeclaration(recordDecl(hasAnyName(
```
Comment (tomflinda) [inline clang/lib/DPCT/ASTTraversal.cpp]: Pls format line13102

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 2]
```diff
@@ -13420,7 +13420,7 @@ void TextureRule::runRule(const MatchFinder::MatchResult &Result) {
     if (DpctGlobalInfo::useSYCLCompat())
       return;
     auto BaseTy = DpctGlobalInfo::getUnqualifiedTypeName(
-        ME->getBase()->getType(), *Result.Context);
+        ME->getBase()->getType().getDesugaredType(*Result.Context), *Result.Context);
```
Comment (tangjj11) [inline clang/lib/DPCT/ASTTraversal.cpp]: Need format

File: clang/test/dpct/texture_object_driver.cu [hunk 3]
```diff
@@ -244,3 +244,18 @@ void foo(){
...
 
+// CHECK: typedef dpct::image_data REDEF_CUDA_RESOURCE_DESC;
+// CHECK-NEXT: REDEF_CUDA_RESOURCE_DESC check_typedefed_cuda_type(dpct::image_matrix_p arr, dpct::image_data a) {
+// CHECK-NEXT:   REDEF_CUDA_RESOURCE_DESC rdesc;
+// CHECK-NEXT:   rdesc.set_data_type(dpct::image_data_type::matrix);
+// CHECK-NEXT:   rdesc.set_data_ptr(arr);
+// CHECK-NEXT:   return rdesc;
+// CHECK-NEXT: }
+typedef CUDA_RESOURCE_DESC REDEF_CUDA_RESOURCE_DESC;
+REDEF_CUDA_RESOURCE_DESC check_typedefed_cuda_type(CUarray arr, cudaResourceDesc a) {
```
Comment (tangjj11) [inline clang/test/dpct/texture_object_driver.cu]: seems `a` is no use.

**Review by tangjj11 — APPROVED** [review]

File: clang/lib/DPCT/ASTTraversal.cpp (hunk 2 above)
Comment (the-slow-one) [reply clang/lib/DPCT/ASTTraversal.cpp]: ~`clang-format` got no suggestion. Could you please suggest the formatted code?~

Fixed line too long!

File: clang/lib/DPCT/ASTTraversal.cpp (hunk 1 above)
Comment (the-slow-one) [reply clang/lib/DPCT/ASTTraversal.cpp]: Done

File: clang/test/dpct/texture_object_driver.cu (hunk 3 above)
Comment (the-slow-one) [reply clang/test/dpct/texture_object_driver.cu]: Removed.

**Pull request merged**